client/my-sites/theme/main.jsx

```diff
@@ -971,7 +972,11 @@ class ThemeSheet extends Component {
 				</span>
 			);
 		} else if ( isLoggedIn && siteId ) {
-			if ( isPremium && ! isThemePurchased && ! isExternallyManagedTheme ) {
+			if (
+				( isPremium || isBundledSoftwareSet ) &&
```

Comment:
I changed it as part of this commit https://github.com/Automattic/wp-calypso/pull/84886/commits/cd6f5da9c78e746603edd34085c31e0a36b87b88, so free themes will also ask for upgrade when it has a bundle.

To test it, you can checkout the branch from this PR: https://github.com/Automattic/wp-calypso/pull/85309, and try to install the "Course" theme on a site with the free plan.

Reply:
🤔 I'm afraid I'm still getting 'Activate' rather than "upgrade to activate' in the three dot menu on a Free site, whether on this branch or #85309 PR.

<img width="803" alt="Screenshot 2023-12-18 at 15 15 42" src="https://github.com/Automattic/wp-calypso/assets/93301/0033cb27-c803-4704-a4b8-78f5ec9bae95">

Reply:
oh hmm, you might get different results in development vs wpcalypso (the calypso live links in pr comments) due to some WIP refactoring we have going on.

What happens if you change `config/development.json` to set `themes/tiers: false` in your local dev env? Does this code work?

Reply:
Good catch! I completely missed this one.

The original commit was supposed to fix the button of the following page for free themes with bundle:

<img width="922" alt="Screenshot 2023-12-18 at 17 39 12" src="https://github.com/Automattic/wp-calypso/assets/876340/be74d8c9-7969-4a5d-9f16-07c662edd996">

But while checking it to fix the dropdown link, I noticed that the `isThemePurchased` wasn't correct for this situation, so I also changed that a bit to check if the site has the feature to install plugins.

Just notice that it doesn't "fix" (or change?) the behavior we talked about in p1700684964931779/1700605581.879429-slack-C048CUFRGFQ, that if the site was already upgraded, it only installs the theme.

I'm not super confident about this change, mainly about the `client/my-sites/themes/theme-options.js` that has many checks in the functions `hideForTheme` for different situations. I'm a little afraid I'm missing something.

Reply:
> What happens if you change config/development.json to set themes/tiers: false in your local dev env? Does this code work?

I tested with the feature flag enabled and disabled and it worked for me. I tested with Course (free and bundled), Tsubaki (premium and bundled), a premium theme, and a free theme.

Reply:
> To test it, you can checkout the branch from this PR: https://github.com/Automattic/wp-calypso/pull/85309, and try to install the "Course" theme on a site with the free plan.

BTW, I think it's fine to test this in this branch directly. You just need to go to that branch (already updated with the last commit) if you want to see all the correct messages and complete the whole flow. 😉

Reply:
> I'm not super confident about this change, mainly about the client/my-sites/themes/theme-options.js that has many checks in the functions hideForTheme for different situations. I'm a little afraid I'm missing something.

Yeah, we don't like it either! We'd dearly like to get rid of it when we can.